action.yml

```diff
@@ -37,13 +43,33 @@ runs:
         fi
         set -e
 
-        mkdir -p ${{ inputs.install-dir }}
+        export install_dir="${{ inputs.install-dir }}"
+
+        if [[ ! -d "$RUNNER_TOOL_CACHE" ]]; then
+          echo "Cache directory '$RUNNER_TOOL_CACHE' does not exist" >&2
+          exit 1
+        fi
+
+        if [[ -z "${{ inputs.install-dir }}" ]]; then
+          export install_dir="$RUNNER_TOOL_CACHE/cosign/${{ inputs.cosign-release }}/${{ runner.arch }}"
+
+          if command -v "$install_dir/cosign" &> /dev/null; then
+              echo "Found Cosign in the cache $RUNNER_TOOL_CACHE"
```

Comment:
wdyt about printing the shasum of the binary that is found? I'm wondering about if someone were to "poison" the cache before our action is called, and we're simply just checking that a binary exists in that path instead of ensuring it is the "right" binary.

Reply:
sounds a great idea, will add another output

Reply:
the only other thought I have is to have the action to retain a full list of all checksums, and ensure the digest of the file matches our "database"

Reply:
If that is something we don't want, we can just set it as won't fix it and keep as is

this is most relevant to self hosted runners